SimpleClient/Source/Handler.cpp

```diff
@@ -22,344 +21,417 @@ Handler::Handler()
-				std::string strToParse = _data.substr(findPosZero + 1u, 
-													  foundPosFirst - findPosZero - 1u);
-				bool flag1;
-				result.firstPoint = utils::fromString<RobotData>(strToParse, flag1);
-
-				const std::size_t foundPosSecond = _data.find('|', foundPosFirst + 1u);
-				strToParse = _data.substr(foundPosFirst + 1u, foundPosSecond - foundPosFirst - 1u);
-				bool flag2;
-				result.secondPoint = utils::fromString<RobotData>(strToParse, flag2);
-
-				strToParse = _data.substr(foundPosSecond + 1u);
-				bool flag3;
-				result.numberOfIterations = utils::fromString<int>(strToParse, flag3);
-
-				if (flag1 && flag2 && flag3)
-				{
-					return result;
-				}
-			}
-			break;
+        case State::POSITIONAL:
+        {
+            const long long num = std::count(_data.begin(), _data.end(), '|');
+            if (_data.size() > 1u && num >= 2)
+            {
+                result.points.reserve(num);
 
-		case State::HOME:
-			[[fallthrough]];
-		case State::FROM_FILE:
-			if (_data.size() == 1u)
-			{
-				return result;
-			}
-			break;
+                std::size_t prev = 0;
 
-		case State::FORWARD:
-			[[fallthrough]];
-		case State::BACK:
-			[[fallthrough]];
-		case State::LEFT:
-			[[fallthrough]];
-		case State::RIGHT:
-			[[fallthrough]];
-		case State::UP:
-			[[fallthrough]];
-		case State::DOWN:
-			[[fallthrough]];
-		case State::ROLL_PLUS:
-			[[fallthrough]];
-		case State::ROLL_MINUS:
-			[[fallthrough]];
-		case State::PITHCH_PLUS:
-			[[fallthrough]];
-		case State::PITHCH_MINUS:
-			[[fallthrough]];
-		case State::YAW_PLUS:
+                std::size_t count = 0;
+
+                bool flag = true;
+
+                _data += '|';
+
+                for (std::size_t i = 0; i < _data.size(); ++i)
+                {
+                    if (_data[i] == '|')
+                    {
+                        const std::string strToParse = _data.substr(prev, i - prev);
+
+                        prev = i + 1;
+
+                        ++count;
+
+                        bool locFlag = true;
+
+                        if (count >= 2)
+                        {
+                            result.points.emplace_back(utils::fromString<RobotData>(strToParse,
+                                locFlag));
+                        }
+
+                        flag &= locFlag;
+                    }
+                }
+
+                if (flag)
+                {
+                    return result;
+                }
+            }
+
+            break;
+        }
+
+		case State::CIRCLIC:
 			[[fallthrough]];
-		case State::YAW_MINUS:
-			if (_data.size() > 1u)
-			{
-				if (const int coefficient = utils::stringToInt(_data.substr(1u)); coefficient > 0)
-				{
-					result.coefficient = coefficient;
-					setCoefficient(coefficient);
-					return result;
-				}
-
-				_state = State::DEFAULT;
-			}
-			else
+		case State::PARTIAL:
+		{
+            const long long num = std::count(_data.begin(), _data.end(), '|');
```

Comment:
**(note):** May be you should use unsigned ll? Because std::count return number of elements, it can't be negative.

Reply:
Heh. I often say: "Always read documentation".
http://en.cppreference.com/w/cpp/algorithm/count